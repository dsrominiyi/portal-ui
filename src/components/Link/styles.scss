@import '../../style/_vars';

.link {
  color: $brand-mid-grey;
  cursor: pointer;
  text-decoration: none;

  &:hover {
    color: $brand-mid-grey;
    text-decoration: underline;
  }
}
