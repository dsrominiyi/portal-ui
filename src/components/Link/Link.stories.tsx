import React from 'react';
import { storiesOf } from '@storybook/react';

import Link from '.';

import { customLink } from '../../../.storybook/styles.scss';

const stories = storiesOf('Link', module);

const render = (
  label: string,
  href?: string,
  classNames?: string[],
  onClick: () => void = () => alert('You clicked!')
) => () => (
  <Link
    classNames={classNames}
    label={label}
    href={href}
    newTab={!!href}
    onClick={href ? undefined : onClick}
  />
);

export const storyRenders = {
  withHref: render('Google', 'https://www.google.com'),
  withOnClick: render('Click me!'),
  customStyling: render('Click me!', undefined, [customLink])
};

stories.add('With href', storyRenders.withHref);
stories.add('With onClick function', storyRenders.withOnClick);
stories.add('With custom styling', storyRenders.customStyling);
