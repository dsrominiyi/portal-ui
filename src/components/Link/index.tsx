import React, { FunctionComponent, ReactNode, MouseEvent } from 'react';

import styles from './styles.scss';

interface Props {
  label: ReactNode;
  icon?: ReactNode;
  href?: string;
  newTab?: boolean;
  classNames?: string[];
  onClick?(event: MouseEvent): void;
}

export const Link: FunctionComponent<Props> = ({
  label,
  icon,
  href,
  newTab,
  classNames,
  onClick
}: Props) => (
  <a
    className={classNames ? classNames.join('') : styles.link}
    href={href}
    target={!onClick && newTab ? '_blank' : undefined}
    onClick={onClick}
  >
    {icon}
    {label}
  </a>
);

Link.displayName = 'Link';
export default Link;
