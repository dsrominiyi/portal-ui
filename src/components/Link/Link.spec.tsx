import { storyRenders } from './Link.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('Link', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
