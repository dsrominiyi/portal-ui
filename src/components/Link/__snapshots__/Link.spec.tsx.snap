// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component story renders match snapshots: customStyling 1`] = `
<a
  className="customLink"
  onClick={[Function]}
>
  Click me!
</a>
`;

exports[`Link component story renders match snapshots: withHref 1`] = `
<a
  className="link"
  href="https://www.google.com"
  target="_blank"
>
  Google
</a>
`;

exports[`Link component story renders match snapshots: withOnClick 1`] = `
<a
  className="link"
  onClick={[Function]}
>
  Click me!
</a>
`;
